### PR TITLE
As an interviewer, I want to be able to start an interview

### PR DESCRIPTION
**Pull request by vvs79**
code review:
- [ ] @DimaSmagulov 
- [ ] @krisberry 
- [ ] @natalya5 
- [ ] @oleg08 
- [x] @OrestF 
- [ ] @rtershak 
- [ ] @CHILL99 

2nd lvl review:

- [ ] @Kmalynss
- [ ] @serhiym 
- [ ] @ruba-ruba

### CODE REVIEW

File: app/controllers/interviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+class InterviewsController < ApplicationController
+	before_action :authenticate_user!
+  before_action :check_interviewer
```
Comment (OrestF) [inline app/controllers/interviews_controller.rb]: indentation

File: app/models/interview_question.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class InterviewQuestion < ActiveRecord::Base
+	belongs_to :interview
+end
```
Comment (OrestF) [inline app/models/interview_question.rb]: indentation

File: app/views/interviews/passage.html.slim [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      - num=0
+      - arr_q=[]
+      - arr_a=[]
+      - @questions.each do |q|
+        - arr_q.push(q.content)
+        - arr_a.push(q.answer)
+        - num=num+1
+        - l="li"+num.to_s
+        li id=l
+          - hrf="#tabs-"+num.to_s
```
Comment (OrestF) [inline app/views/interviews/passage.html.slim]: I think you should to use helper here

Comment (vvs79) [inline app/views/interviews/passage.html.slim]: helper for ruby-code ?

File: app/assets/stylesheets/application.css.sass [hunk 4]
```diff
@@ -1,15 +0,0 @@
...
- * or any plugin's vendor/assets/stylesheets directory can be referenced here using a relative path.
- *
- * You're free to add application-wide styles to this file and they'll appear at the bottom of the
- * compiled file so the styles you add here take precedence over styles defined in any styles
- * defined in the other CSS/SCSS files in this directory. It is generally better to create a new
- * file per style scope.
- *
- *= require_tree .
- *= require_self
- */
```
Comment (krisberry) [inline app/assets/stylesheets/application.css.sass]: Why did you remove this file?

**Comment by DimaSmagulov** [comment]
pls add error and flash messages in your views

File: app/controllers/interviews_controller.rb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  add_breadcrumb "interviews", :interviews_path
+  def index
+    @interviews = Interview.paginate(page: params[:page], :per_page => 5)
+  end
+	def new
+    @interview = Interview.new
+    add_breadcrumb "new_interview", new_interview_path
+  end
+
+  def show
```
Comment (DimaSmagulov) [inline app/controllers/interviews_controller.rb]: pls add rescue for ActiveRecord

File: app/assets/stylesheets/application.css.sass (hunk 4 above)
Comment (DimaSmagulov) [inline app/assets/stylesheets/application.css.sass]: I think he didn't.

Comment (krisberry) [inline app/assets/stylesheets/application.css.sass]: I pulled this branch to ensure I wasn't mistaken and I didn't find this file. But right now you are right, it's present here =)

File: app/controllers/interviews_controller.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    @interview = Interview.find(params[:id])
+    add_breadcrumb 'passage_interview_'+(@interview.id).to_s, passage_interview_path(@interview)
+    @id_questions = []
+    @questions = @interview.template.questions
+    @all_questions = Question.all
+    @all_questions.each do |aq|
+      @questions.each do |q|
+        @id_questions.push(aq.id)
+        InterviewQuestion.create(question_id: aq.id, interview_id: @interview.id) if aq.content == q.content
+      end
```
Comment (OrestF) [inline app/controllers/interviews_controller.rb]: I did not fully understand, why you use @all_questions?

Comment (vvs79) [inline app/controllers/interviews_controller.rb]: I use @all_questions to find out question.id in model Question

Comment (OrestF) [inline app/controllers/interviews_controller.rb]: In this case we iterate through all questions in DB and its bad for performance. 
Maybe try to  you use @tempalte.questions instead?

File: app/helpers/interviews_helper.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+module InterviewsHelper
+  def email_user(user_id)
```
Comment (ruba-ruba) [inline app/helpers/interviews_helper.rb]: this method needs to be reworked

File: app/assets/stylesheets/interviews.scss [hunk 8]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the interviews controller here.
+// They will automatically be included in application.css.
```
Comment (ruba-ruba) [inline app/assets/stylesheets/interviews.scss]: if there no any css, don't include such files

File: app/controllers/interviews_controller.rb [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  end
+
+  def destroy
+    Interview.find(params[:id]).destroy
+    flash[:success] = 'Interview deleted'
+    redirect_to :back
+  end
+
+  def passage
```
Comment (Kmalynss) [inline app/controllers/interviews_controller.rb]: What is ```passage``` and how it is related to your task?

File: app/views/interviews/passage.html.slim [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    ul
+      - num=0
+      - arr_q=[]
+      - arr_a=[]
+      - @questions.each do |q|
+        - arr_q.push(q.content)
+        - arr_a.push(q.answer)
+        - num+=1
+        - l="li"+num.to_s
+        - hrf="#tabs-"+num.to_s
```
Comment (Kmalynss) [inline app/views/interviews/passage.html.slim]: Such code is not allowed in views.

File: app/views/interviews/passage.html.slim [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        li.d-line id=l
+          a href=hrf
+            = num
+    br
+    - num_tab=0        
+    - num.times do
+      - hrf="tabs-"+(num_tab+1).to_s
```
Comment (Kmalynss) [inline app/views/interviews/passage.html.slim]: Such code is not allowed in views.

File: app/views/interviews/passage.html.slim [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+          = num_tab+1
+          | :  
+          = arr_q[num_tab]
+        h4
+          | Answer: 
+          = arr_a[num_tab]
+        br
+        h4
+          | Comment: 
+        = text_area(:text, "area-"+(num_tab+1).to_s, placeholder: "Write a comment", rows: 6)
```
Comment (Kmalynss) [inline app/views/interviews/passage.html.slim]: Such code is not allowed in views.

**Comment by serhiym** [comment]
maybe easier to start this task from scratch )) ?

File: app/helpers/interviews_helper.rb [hunk 13]
```diff
@@ -0,0 +1,8 @@
...
+    users=User.all
+    email=''
+    users.each { |user| email=user.email if user.id==user_id }
+    user_email=email
+  end
+end
```
Comment (Kmalynss) [inline app/helpers/interviews_helper.rb]: @vvs79 
There are 2 things about this helper:
1. It is totally not needed.
2. It is really very and very bad.

Please explain both statements.

File: app/assets/javascripts/passage.js [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+  $("input[name='mark']").click(function() {
+    var id_radio = $(this).attr("id");
+    var color_radio = id_radio.charAt(5);
+    if (color_radio=='g') { arr_marks[parseInt(id_radio.slice(11))-1]='green'; color='green'; }
+    if (color_radio=='y') { arr_marks[parseInt(id_radio.slice(12))-1]='yellow'; color='yellow'; }
+    if (color_radio=='r') { arr_marks[parseInt(id_radio.slice(9))-1]='red'; color='red'; }
+    $("a#ui-id-"+num).css("background",color);
+  });
+
+});
```
Comment (Kmalynss) [inline app/assets/javascripts/passage.js]: What is this?
Why do we need it?
Is it really related to this task?

File: app/views/interviews/_interview.html.slim [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+        = interview.firstname
+    td
+      = link_to interview_path(interview)
+        = interview.lastname
+    td
+      = link_to interview_path(interview)
+        = interview.updated_at.to_s[0..15]
+    td
+      = link_to interview_path(interview), method: :delete, data: { confirm: 'Are you sure?' }, class: "btn btn-sm btn-danger" do
+        i.glyphicon.glyphicon-remove Delete
```
Comment (Kmalynss) [inline app/views/interviews/_interview.html.slim]: Why each line here is a link to a interview?

**Comment by Kmalynss** [comment]
Please given more meaningful names to PRs.

**Comment by vvs79** [comment]
https://trello.com/c/VxoE83hC/50-as-an-interviewer-i-want-to-be-able-to-start-an-interview

**Comment by vvs79** [comment]
- [ ] Mercury
- [x] Venus
- [x] Earth (Orbit/Moon)
- [x] Mars
- [ ] Jupiter
- [ ] Saturn
- [ ] Uranus
- [ ] Neptune
- [ ] Comet Haley
code review:

 @DimaSmagulov
 @krisberry
 @natalya5
 @oleg08
 @OrestF
 @rtershak
 @oleg08
2nd lvl review:

 @Kmalynss
 @serhiym
 @ruba-ruba

**Comment by vvs79** [comment]
code review:

 - [ ] @DimaSmagulov
 - [ ] @krisberry
 - [ ] @natalya5
 - [ ] @oleg08
 - [ ] @OrestF
 - [ ] @rtershak
 - [ ] @ole08

2nd lvl review:

- [ ]  @Kmalynss
 - [ ] @serhiym
 - [ ] @ruba-ruba

**Comment by vvs79** [comment]
code review:

 - [ ] @DimaSmagulov
 - [ ] @krisberry
 - [ ] @natalya5
 - [ ] @oleg08
 - [ ] @OrestF
 - [ ] @rtershak
 - [ ] @CHILL99 
2nd lvl review:

 - [ ] @Kmalynss
 - [ ] @serhiym
 - [ ] @ruba-ruba

**Comment by vvs79** [comment]
code review:
- [ ] @DimaSmagulov 
- [ ] @krisberry 
- [ ] @natalya5 
- [ ] @oleg08 
- [ ] @OrestF 
- [ ] @rtershak 
- [ ] CHILL99 

2nd lvl review:

- [ ] @Kmalynss
- [ ] @serhiym 
- [ ] @ruba-ruba

**Comment by Kmalynss** [comment]
@vvs79 please add tests
